encode.go

```diff
@@ -63,7 +63,16 @@ func encodeSQLValue(w io.Writer, arg interface{}) error {
 		// We may be in the presence of a type alias not supported by the
 		// database/driver DefaultParameterConverter. Special handling.
 		value := reflect.ValueOf(arg)
-		if baseKinds[value.Kind()] {
+		kind := value.Kind()
+		// try uint64 with high bit set first to avoid infinite recursion.
+		// 9223372036854775808 is the lowest uint64 with high bit set.
+		if kind == reflect.Uint64 && value.Uint() >= 9223372036854775808 {
```

Comment:
You should use `>= 1<<63` so that you can you can see at a glance that this is correct.

Reply:
Maybe uint(1) << 63?

Reply:
fixed

Reply:
going to circle back to handle both `uint64` and `uint` and their pointers

Reply:
ok, pushed once more—maybe for the last time

Reply:
yes, that's pretty bad, `defaultConverter.ConvertValue` does not behave consistently for high-bit-set `uint64` versus `uint`.  And if they "fixed" that then the `uint` flavor would start erroring too.
this issue would be protected against now that this PR's special casing has been moved above the default converter.